TCAT/Controllers/RouteOptionsViewController.swift

```diff
@@ -298,25 +302,74 @@ class RouteOptionsViewController: UIViewController, UITableViewDelegate, UITable
             // Prepare feedback on Network request
             let tapticGenerator = UIImpactFeedbackGenerator(style: .medium)
             tapticGenerator.prepare()
+            
+            let sameLocation = searchFrom.name == searchTo.name
+            
+            /// Determine if coordinates in parameters are within range
+            func validCoordinates(_ requestParameters: [String : Any]?) -> Bool {
```

Comment:
get rid of nested functions.  hard to read. Also why do you check validCoordinates from the `requestParameters` and not the `searchTo` and `searchFrom` values themselves? What edge case is this suppose to handle?

Reply:
We don't know the coordinates for google places until right before the request. It needs to be looked up via a Google API call. In that callback, we do the getRoutes network call, so once I have all the coordinates I check with the parameters generated. Given the current functions and classes that's the most accessible way to get the coords. I'd love to refactor this, I think it might sense to have one Place type and a variable called `type` that signifies whether it's a Google Place or Bus Stop. Then we can store coordinates once we look up the Google Place. Just one idea

Reply:
Okay I think I can refactor it to be better.

Jw do you know any coordinates that are invalid? Surprised that Google maps doesn’t return valid coordinates all the time even tho it’s Google